Anlab.Mvc/Controllers/PaymentController.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AnlabMvc.Models.Configuration;
+using Anlab.Core.Data;
+using AnlabMvc.Models.Order;
+using AnlabMvc.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace AnlabMvc.Controllers
+{
+    public class PaymentController : ApplicationController
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IDataSigningService _dataSigningService;
+        private readonly AppSettings _appSettings;
+        private readonly CyberSourceSettings _cyberSourceSettings;
+
+        public PaymentController(ApplicationDbContext context, IDataSigningService dataSigningService, IOptions<CyberSourceSettings> cyberSourceSettings, IOptions<AppSettings> appSettings)
+        {
+            _context = context;
+            _dataSigningService = dataSigningService;
+            _appSettings = appSettings.Value;
+            _cyberSourceSettings = cyberSourceSettings.Value;
+        }
+
+        public ActionResult Pay(int id)
+        {
+            var order = _context.Orders.SingleOrDefault(o => o.Id == id);
+
+            if (order == null)
+            {
+                return NotFound(id);
+            }
+            if (order.CreatorId != CurrentUserId)
+            {
+                ErrorMessage = "You don't have access to this order.";
+                return RedirectToAction("Index", "Order");
+            }
+
+            if (order.Status != OrderStatusCodes.Complete)
+            {
+                ErrorMessage = "You can't edit an order that has been confirmed.";

```

Comment:
@jSylvestre We might want a different status in here but for now completed is ok.  Afterwards we can move to a "Paid" status or similar.  Probably will require changing this error message though because the edit text doesn't make sense